x-pack/plugins/asset_manager/server/lib/implicit_collection/collector_runner.ts

```diff
@@ -19,15 +24,19 @@ export class CollectorRunner {
   }
 
   async run() {
-    const collectorOptions = {
-      client: this.options.inputClient,
-      from: Date.now() - this.options.intervalMs,
-    };
+    const now = Date.now();
 
     for (let i = 0; i < this.collectors.length; i++) {
```

Comment:
note: sequential executions is used for simplicity. we could try to naively parallelize execution of the queries but a streaming implementation should be more sane/tunable in that case. Each collector can implement readable and produce assets on each pagination, with a downstream writable that bulk writes to elasticsearch and is used as backpressure